src/send/mod.rs

```diff
@@ -75,8 +76,11 @@ fn validate(email: &Email) -> bool {
     true
 }
 
-fn fail() -> data::Outcome<Email, ()> {
-    Outcome::Failure((Status::BadRequest, ()))
+fn fail() -> data::Outcome<Email, HandlerError> {
+    Outcome::Failure((
+        Status::BadRequest, 
+        HandlerErrorKind::InvalidEmailAddress.into()
```

Comment:
This probably shouldn't be hard-coded at this point because `fail` is also called if the `provider` param is invalid. And maybe it will be called for other params in the future, too. Let's either rework the outer logic so the `HandlerErrorKind` can be passed in as an argument, or keep that the same and change this to a more generic validation error that doesn't mention email addresses.